modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -80,213 +139,233 @@
 		forge_temperature += 5
 
 	if(forge_temperature > 0)
-		if(primitive)
-			icon_state = "primitive_forge_full"
-		else
-			icon_state = "forge_full"
+		icon_state = "[primitive ? "primitive_" : ""]forge_full"
 		light_range = 3
 	else if(forge_temperature <= 0)
-		if(primitive)
-			icon_state = "primitive_forge_empty"
-		else
-			icon_state = "forge_empty"
+		icon_state = "[primitive ? "primitive_" : ""]forge_empty"
 		light_range = 0
 
+/**
+ * Here we fix any weird in_use bugs
+ */
 /obj/structure/reagent_forge/proc/check_in_use()
 	if(!in_use)
 		return
 	for(var/mob/living/living_mob in range(1,src))
 		if(!living_mob)
 			in_use = FALSE
 
+/**
+ * Here we spawn coal depending on a chance of using wood
+ */
 /obj/structure/reagent_forge/proc/spawn_coal()
 	new /obj/item/stack/sheet/mineral/coal(get_turf(src))
 
 /obj/structure/reagent_forge/process()
-	if(world_check >= world.time) //to make it not too intensive, every 5 seconds
+	if(!COOLDOWN_FINISHED(src, forging_cooldown)) //every 5 seconds to not be too intensive, also balanced around 5 seconds
 		return
-	world_check += 5 SECONDS
+	COOLDOWN_START(src, forging_cooldown, 5 SECONDS)
 	check_fuel()
 	check_temp()
 	check_in_use() //plenty of weird bugs, this should hopefully fix the in_use bugs
 
+/obj/structure/reagent_forge/attack_hand(mob/living/user, list/modifiers)
```

Comment:
For all these upgrades, it might be better to flavor it as 'looking at [x part to change], you see room for improvement, and adjust accordingly' or 'watching [x part to change] you notice [x], and modify the forge to avoid this!'

It makes it seem like you're actually learning the forge in and out, rather than just placing your hand on it to upgrade it